### ISSUE TITLE
"React Native version mismatch"

### ISSUE THREAD

**Issue by Sergeypt423** [expo/expo#923]
I've used react-native@0.50.1 and expo@22.0.2 and get this error.

"React Native version mismatch 

JavaScript version: 0.50.1
Native version: 0.49.3"

How I can solve this?
Which version of expo support react-native@0.51.1?

**Comment by Balasnest** [comment]
Me too getting same issue.

**Comment by SiLenTTz** [comment]
Same issue here. The error on the device says, that we have Native version 0.49.0 and JS version 0.50.1. When calling 'react-native -version' in command line on Windows it says version 0.50.1 is currently installed

**Comment by shmox** [comment]
Same error here

JavaScript version: 0.49.5
Native version: 0.50.1

**Comment by dikaiosune** [comment]
Hi! I would recommend double checking your package.json file to make sure you're using a compatible version of RN. We currently support RN 0.49.*, and expect to have a release supporting 0.50 very soon!

**Comment by SiLenTTz** [comment]
In the Json file the correct Version ist displayed but maybe not yet supported as you said. Then there is no possibility to use the camera, right?

**Comment by carloshpdoc** [comment]
I upgraded the Nodejs and NPM to the Latest LTS Version: 8.9.0 (including NPM 5.5.1) and it worked for me.

**Comment by diamedi** [comment]
I had version: "react-native": "0.50.2"
I used the command:
npm install react-native@0.49
I think it took it down to react-native@0.49.5
I didn't get the version mismatch error anymore so it worked for me :)

**Comment by ide** [comment]
Hi everyone - just for clarity:

Expo SDK 22 uses RN 0.49
Expo SDK 23 will use RN 0.50

And to find this out in general, if you're using the Expo fork of RN (recommended) which has an SDK version number in the .tar.gz URL, you can look at `node_modules/react-native/package.json` to see the upstream RN version number.

It is important that your JS matches the native code. RN is not reliably backwards or forwards compatible so make sure your versions match.

**Comment by afilp** [comment]
I have an ejected project and I still get this error. 

How can I get rid of 0.49, in favor of 0.50.3 that I am using in package.json?

I am also in Windows and not sure how to clean the "maven" repository, if this is needed.

Thanks!

**Comment by aithashi** [comment]
Sometimes it also happens when you try to run without closing the node server, in which the previous app was running, so try closing the node server and start fresh again!

**Comment by arnoldbird** [comment]
In this context, what does "Javascript version" and "Native version" mean?  Uh, I mean, what is mismatched with what?

I ran the suggested watchman command, to no avail.

I don't specify any RN version in my package.json.  Instead I have...

`"react-native": "https://github.com/expo/react-native/archive/sdk-26.0.0.tar.gz",`

Previously, I had...

`"react-native": "https://github.com/expo/react-native/archive/sdk-25.0.0.tar.gz",`

**Comment by simonbw** [comment]
@arnoldbird I had the same problem. I fixed it by changing my react-native dependency to:

 ```"react-native": "^0.54.2",```

My expo dependency is

```"expo": "^26.0.0",```

**Comment by arnoldbird** [comment]
@simonbw 

I think that's basically it.  I messed up a few different things at the same time.  One problem was that my expo version did not match the react-native version in my package.json.  

I was tripping over "Native" above, which I now understand means "React Native".  I had just never seen it shortened that way.  It's always "RN" or "react".   I was thinking "Native" was something in the simulator code.  Glad there is actually no new concept to learn.

My various problems are all resolved and my app is working again.

**Comment by danlin604** [comment]
I tried changing dependency, rebooting my phone, deleting yarn.lock & node_module, killing my node processes. None of these fixes the version mismatch issue. The weirdest part is that the issue vanished from my Android phone, but appears in for other iOS users trying to access my local project.

**Comment by EvanBacon** [comment]
@danlin604 please make sure your computer is plugged in 😂😝
I run into this error sometimes, looks like you did everything right, I would also try these:
* kill the expo client on your iPhone
* run `exp start -c` the `-c` will clear caches
💙💙💙

**Comment by danlin604** [comment]
Thanks for the response. I tried `exp start -c` a few times. The funny thing is that now my Android is now experiencing the mismatch issue. I am going to re-clone the project, clear everything, wipe temp file, re-install XDE, and restart my system.

**Comment by EvanBacon** [comment]
@danlin604  Sorry you are having these problems. This stuff is related to the `node_modules` and the version number in your `app.json` so you shouldn't have to do much more than editing `app.js` and `package.json`

**Comment by sojimon** [comment]
@jeevantakhar  When I run `watchman watch-del-all && react-native start --reset-cache` metro bundler runs.

**Comment by pashute** [comment]
I didn't have watchman.

`react-native info`:
    
  Environment:
  OS: Windows 10
  Node: 8.11.1
  Yarn: 1.5.1
  npm: 6.0.0
  Watchman: Not Found
  Xcode: N/A
  Android Studio: Version  3.1.0.0 AI-173.4670197

  Packages: (wanted => installed)
  react: 16.3.1 => 16.3.1
  react-native: 0.55.3 => 0.55.3

**Comment by seenickcode** [comment]
@pashute, could you provide the commands you've used?

**Comment by pashute** [comment]
My old project didn't work. Got this message no matter what I did. So I created a new project with react-native init, copied my project specific files to it and then after the list of npm installs and updating the project.json and app.json all worked ok. I can't go back now, because I also updated everything and am now working with the latest version of everything.

**Comment by seenickcode** [comment]
Hmm ok.
 
I decided to skip Expo altogether myself and use react-native init. 

Good article here I wish I read before starting re considerations on when to use Expo and when to not:  https://medium.com/@paulsc/react-native-first-impressions-expo-vs-native-9565cce44c92

**Comment by Pines-Cheng** [comment]
How can I get the Corresponding  React Native version of Expo?

**Comment by pashute** [comment]
react-native info

בתאריך יום ד׳, 18 ביולי 2018, 09:57, מאת Pines ‏<notifications@github.com>:

**Comment by roktoshl** [comment]
I had to run `react-native start --reset-cache`

**Comment by vkumarNobul** [comment]
Please check if you are running "npm start -- --reset-cache" for another project.

**Comment by musaatalay1** [comment]
check  expo version in both of app.json and package.json, they both must match. in app.json it is defined as sdkVersion.

**Comment by TheSalarKhan** [comment]
Yep it was the app.json for me. Just had to change the sdkVersion there apart from updating package.json

**Comment by wwwhatley** [comment]
@musaatalay1 This worked for me! Thank you 👍

**Comment by vonwao** [comment]
I'm having a similar issue now with the new expo SDK (33 vs 32).

**Comment by jcush** [comment]
For anyone who Googles their way here looking for similar answers, if your version mismatch shows a higher Native version than JavaScript version, the Expo app on the simulator/device is the issue. `expo-cli` should automatically update simulator apps, but if you don't have it installed then this obviously won't happen.

In this case:
- on a physical device just download the latest version of the Expo app
- on a simulator, delete the Expo app and re-run the project on the simulator from your expo development server (either from the page which launches in your browser, or from the CLI using `a` for Android or `i` for iOS

This solved it for me anyway, hope this helps!

@vonwao shot in the dark that this is the fix for you too but maybe?

**Comment by vonwao** [comment]
@jcush Thanks, I believe this solved it for me too.

**Comment by ide** [comment]
If you encounter this issue, you likely need a different version of the react-native package installed. If you use Yarn, you can run `yarn why react-native` to see all the copies installed -- there should be just one -- and its version. If the version of the package installed doesn't match the version the client needs, that's the problem.

Additionally, be sure to restart your Metro bundler when installing new npm packages since it may not pick up the newly installed version automatically.

I'm going to lock this issue since the original issue was opened a year and a half ago. Even if there are issues today with the same symptom, the causes are often likely to be different. GitHub issues are often most effective when they are organized by cause rather than effect, so if this recurs, a separate issue will help keep the signal high.
